cs_upload.py

```diff
@@ -203,14 +203,25 @@ def upload_to_s3(self, form_data: str, trace_file: str) -> None:
         fields["success_action_status"] = str(fields["success_action_status"])
         fields["x-amz-meta-filename"] = basename(trace_file)
 
+        (_, extension) = splitext(trace_file)
+        if extension == ".gz":
+            mime_type = "application/gzip"
+        elif extension == ".cst":
+            mime_type = "application/json"
```

Comment:
It does make sense. It was the absence of a `Content-Type` form element in the query that caused @Nemeras 's original problem, which is why I'm adding it in this MR. I'm sure we didn't need it before. Still don't on Minio. I'm not sure I understand fully, but I think it might somehow be related to the changes to our cross-site settings. I've followed your suggestion and make it conform to those instructions. (I'd forgotten about them, but they're exactly what I needed here.)

Reply:
By the way, fair point about CST files not being strictly JSON: I'd forgotten that.

Reply:
> I think it might somehow be related to the changes to our cross-site settings

I don't see how since those changes only apply to HTML pages as far as I can tell.